graphql/executor.go

```diff
@@ -170,15 +170,17 @@ func PrepareQuery(ctx context.Context, typ Type, selectionSet *SelectionSet) err
 			}
 
 			// Only parse args once for a given selection.
-			if !selection.parsed {
+			if selection.UnparsedArgs == nil {
```

Comment:
Is it possible for the args to be nil? Might make sense to keep the parsed flag.

In which case we could set the "unparsedArgs" wherever we're setting the "args" right now.

So we'd do
```
parsed, err := field.ParseArguments(selection.UnparsedArgs)
```
(which I think is a bit cleaner anyway)

Reply:
thanks, i took your suggestion here.